### ISSUE TITLE
GO Marshall error while ACME challenge on Netcup DNS

### ISSUE THREAD

**Issue by hendriku** [containous/traefik#4125]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I am currently trying to get a wildcard certificate for my traefik system. My Domain is being hosted at netcup. I added the correct ENV variables from the [provider table](https://docs.traefik.io/configuration/acme/#provider) to my traefik container.

### What did you expect to see?

I'd expect the ACME challenge to succeed and Let's Encrypt to provide me a valid certificate.

### What did you see instead?

An error message in the traefik logs and the traefik default cert instead of a valid one.

```
time="2018-10-28T15:02:20Z" level=error msg="Unable to obtain ACME certificate for domains \"*.mydomain.de,mydomain.de\" : unable to generate a certificate for the domains [*.mydomain.de mydomain.de]: acme: Error -> One or more domains had a problem:\n[mydomain.de] error presenting token: netcup: error decoding response of DNS-API, json: cannot unmarshal string into Go struct field ResponseMsg.responsedata of type netcup.ResponseData\n"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.3
Codename:     maroilles
Go version:   go1.11.1
Built:        2018-10-15_10:13:00AM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
defaultEntryPoints = ["http", "https"]
debug = true

[accessLog]
  filePath = "/etc/traefik/traefik-access.log"

[traefikLog]
  filePath = "/etc/traefik/traefik.log"

[docker]
  endpoint = "unix:///var/run/docker.sock"
  domain = "mydomain.de"
  exposedByDefault = false
  watch = true
  network = "traefik"

[entryPoints]
  [entryPoints.http]
    address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"

  [entryPoints.https]
    address = ":443"
    [entryPoints.https.tls]

[acme]
  email = "my@email.com"
  storage = "acme.json"
  entryPoint = "https"
  [acme.dnsChallenge]
    provider = "netcup"
    delayBeforeCheck = 0
  [[acme.domains]]
    main = "*.mydomain.de"
    sans = ["mydomain.de"]
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
root@myhost:/var/lib/docker/volumes/traefik/_data# cat traefik.log
time="2018-10-28T15:19:40Z" level=info msg="Using TOML configuration file /etc/traefik/traefik.toml"
time="2018-10-28T15:19:40Z" level=info msg="Traefik version v1.7.3 built on 2018-10-15_10:13:00AM"
time="2018-10-28T15:19:40Z" level=debug msg="Global configuration loaded {\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"GraceTimeOut\":0,\"Debug\":true,\"CheckNewVersion\":true,\"SendAnonymousUsage\":false,\"AccessLogsFile\":\"\",\"AccessLog\":{\"file\":\"/etc/traefik/traefik-access.log\",\"format\":\"common\"},\"TraefikLogsFile\":\"\",\"TraefikLog\":{\"file\":\"/etc/traefik/traefik.log\",\"format\":\"common\"},\"Tracing\":null,\"LogLevel\":\"\",\"EntryPoints\":{\"http\":{\"Address\":\":80\",\"TLS\":null,\"Redirect\":{\"entryPoint\":\"https\"},\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}},\"https\":{\"Address\":\":443\",\"TLS\":{\"MinVersion\":\"\",\"CipherSuites\":null,\"Certificates\":null,\"ClientCAFiles\":null,\"ClientCA\":{\"Files\":null,\"Optional\":false},\"DefaultCertificate\":null,\"SniStrict\":false},\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}}},\"Cluster\":null,\"Constraints\":[],\"ACME\":{\"Email\":\"my@email.com\",\"Domains\":[{\"Main\":\"*.mydomain.de\",\"SANs\":[\"mydomain.de\"]}],\"Storage\":\"acme.json\",\"StorageFile\":\"\",\"OnDemand\":false,\"OnHostRule\":false,\"CAServer\":\"https://acme-v02.api.letsencrypt.org/directory\",\"EntryPoint\":\"https\",\"KeyType\":\"\",\"DNSChallenge\":{\"Provider\":\"netcup\",\"DelayBeforeCheck\":0},\"HTTPChallenge\":null,\"TLSChallenge\":null,\"DNSProvider\":\"\",\"DelayDontCheckDNS\":0,\"ACMELogging\":false,\"OverrideCertificates\":false,\"TLSConfig\":null},\"DefaultEntryPoints\":[\"http\",\"https\"],\"ProvidersThrottleDuration\":2000000000,\"MaxIdleConnsPerHost\":200,\"IdleTimeout\":0,\"InsecureSkipVerify\":false,\"RootCAs\":null,\"Retry\":null,\"HealthCheck\":{\"Interval\":30000000000},\"RespondingTimeouts\":null,\"ForwardingTimeouts\":null,\"AllowMinWeightZero\":false,\"Web\":null,\"Docker\":{\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"unix:///var/run/docker.sock\",\"Domain\":\"mydomain.de\",\"TLS\":null,\"ExposedByDefault\":false,\"UseBindPortIP\":false,\"SwarmMode\":false,\"Network\":\"traefik\"},\"File\":null,\"Marathon\":null,\"Consul\":null,\"ConsulCatalog\":null,\"Etcd\":null,\"Zookeeper\":null,\"Boltdb\":null,\"Kubernetes\":null,\"Mesos\":null,\"Eureka\":null,\"ECS\":null,\"Rancher\":null,\"DynamoDB\":null,\"ServiceFabric\":null,\"Rest\":null,\"API\":null,\"Metrics\":null,\"Ping\":null,\"HostResolver\":null}"
time="2018-10-28T15:19:40Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
time="2018-10-28T15:19:40Z" level=debug msg="Setting Acme Certificate store from Entrypoint: https"
time="2018-10-28T15:19:40Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redirect:0xc0003e7380 Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc00058b340} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-10-28T15:19:40Z" level=info msg="Preparing server https &{Address::443 TLS:0xc0002a7200 Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc00058b3a0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-10-28T15:19:40Z" level=info msg="Starting server on :80"
time="2018-10-28T15:19:40Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
time="2018-10-28T15:19:40Z" level=info msg="Starting server on :443"
time="2018-10-28T15:19:40Z" level=info msg="Starting provider *docker.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"unix:///var/run/docker.sock\",\"Domain\":\"mydomain.de\",\"TLS\":null,\"ExposedByDefault\":false,\"UseBindPortIP\":false,\"SwarmMode\":false,\"Network\":\"traefik\"}"
time="2018-10-28T15:19:40Z" level=info msg="Starting provider *acme.Provider {\"Email\":\"my@email.com\",\"ACMELogging\":false,\"CAServer\":\"https://acme-v02.api.letsencrypt.org/directory\",\"Storage\":\"acme.json\",\"EntryPoint\":\"https\",\"KeyType\":\"\",\"OnHostRule\":false,\"OnDemand\":false,\"DNSChallenge\":{\"Provider\":\"netcup\",\"DelayBeforeCheck\":0},\"HTTPChallenge\":null,\"TLSChallenge\":null,\"Domains\":[{\"Main\":\"*.mydomain.de\",\"SANs\":[\"mydomain.de\"]}],\"Store\":{}}"
time="2018-10-28T15:19:40Z" level=info msg="Testing certificate renew..."
time="2018-10-28T15:19:40Z" level=debug msg="Looking for provided certificate(s) to validate [\"*.mydomain.de\" \"mydomain.de\"]..."
time="2018-10-28T15:19:40Z" level=debug msg="Configuration received from provider ACME: {}"
time="2018-10-28T15:19:40Z" level=debug msg="Creating entry point redirect http -> https"
time="2018-10-28T15:19:40Z" level=debug msg="Domains [\"*.mydomain.de\" \"mydomain.de\"] need ACME certificates generation for domains \"*.mydomain.de,mydomain.de\"."
time="2018-10-28T15:19:40Z" level=debug msg="Loading ACME certificates [*.mydomain.de mydomain.de]..."
time="2018-10-28T15:19:40Z" level=debug msg="Building ACME client..."
time="2018-10-28T15:19:40Z" level=debug msg="https://acme-v02.api.letsencrypt.org/directory"
time="2018-10-28T15:19:40Z" level=debug msg="Provider connection established with docker 18.06.1-ce (API 1.38)"
time="2018-10-28T15:19:40Z" level=debug msg="Filtering disabled container /traefik"
time="2018-10-28T15:19:40Z" level=debug msg="originLabelsmap[traefik.enable:true traefik.frontend.entryPoints:http,https traefik.frontend.rule:Host:mysubdomain.mydomain.de traefik.port:2368]"
time="2018-10-28T15:19:40Z" level=debug msg="allLabelsmap[:map[traefik.enable:true traefik.frontend.entryPoints:http,https traefik.frontend.mysubdomain.mydomain.de traefik.port:2368]]"
time="2018-10-28T15:19:40Z" level=debug msg="Filtering disabled container /portainer"
time="2018-10-28T15:19:40Z" level=debug msg="originLabelsmap[traefik.enable:true traefik.frontend.entryPoints:http,https traefik.frontend.rule:Host:mysubdomain.mydomain.de traefik.port:2368]"
time="2018-10-28T15:19:40Z" level=debug msg="allLabelsmap[:map[traefik.enable:true traefik.frontend.entryPoints:http,https traefik.frontend.rule:Host:mysubdomain.mydomain.de traefik.port:2368]]"
time="2018-10-28T15:19:40Z" level=debug msg="Backend backend-ghost: no load-balancer defined, fallback to
'wrr' method"
time="2018-10-28T15:19:40Z" level=debug msg="Configuration received from provider docker: {\"backends\":{\"backend-ghost\":{\"servers\":{\"server-ghost-e3d7ca53ad36cdf4885ff43773eccdbf\":{\"url\":\"http://172.18.0.2:2368\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"frontend-Host-mysubdomain-mydomain-de-0\":{\"entryPoints\":[\"http\",\"https\"],\"backend\":\"backend-ghost\",\"routes\":{\"route-frontend-Host-mysubdomain-mydomain-de-0\":{\"rule\":\"Host:mysubdomain.mydomain.de\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
time="2018-10-28T15:19:41Z" level=info msg="Server configuration reloaded on :80"
time="2018-10-28T15:19:41Z" level=info msg="Server configuration reloaded on :443"
time="2018-10-28T15:19:41Z" level=debug msg="Creating entry point redirect http -> https"
time="2018-10-28T15:19:41Z" level=debug msg="Using DNS Challenge provider: netcup"
time="2018-10-28T15:19:41Z" level=debug msg="Wiring frontend frontend-Host-mysubdomain-mydomain-de-0 to entryPoint http"
time="2018-10-28T15:19:41Z" level=debug msg="Creating backend backend-ghost"
time="2018-10-28T15:19:41Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-mysubdomain-mydomain-de-0"
time="2018-10-28T15:19:41Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-28T15:19:41Z" level=debug msg="Creating server server-ghost-e3d7ca53ad36cdf4885ff43773eccdbf at http://172.18.0.2:2368 with weight 1"
time="2018-10-28T15:19:41Z" level=debug msg="Creating route route-frontend-Host-mysubdomain-mydomain-de-0 Host:mysubdomain.mydomain.de"
time="2018-10-28T15:19:41Z" level=debug msg="Wiring frontend frontend-Host-mysubdomain-mydomain-de-0 to entryPoint https"
time="2018-10-28T15:19:41Z" level=debug msg="Creating backend backend-ghost"
time="2018-10-28T15:19:41Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-mysubdomain-mydomain-de-0"
time="2018-10-28T15:19:41Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-28T15:19:41Z" level=debug msg="Creating server server-ghost-e3d7ca53ad36cdf4885ff43773eccdbf at http://172.18.0.2:2368 with weight 1"
time="2018-10-28T15:19:41Z" level=debug msg="Creating route route-frontend-Host-mysubdomain-mydomain-de-0 Host:mysubdomain.mydomain.de"
time="2018-10-28T15:19:41Z" level=info msg="Server configuration reloaded on :80"
time="2018-10-28T15:19:41Z" level=info msg="Server configuration reloaded on :443"
time="2018-10-28T15:19:42Z" level=error msg="Error obtaining certificate retrying in 349.831223ms"
time="2018-10-28T15:19:43Z" level=error msg="Error obtaining certificate retrying in 1.021321236s"
time="2018-10-28T15:19:45Z" level=error msg="Error obtaining certificate: acme: Error -> One or more domains had a problem:\n[mydomain.de] error presenting token: netcup: error decoding response of DNS-API, json: cannot unmarshal string into Go struct field ResponseMsg.responsedata of type netcup.ResponseData\n"
time="2018-10-28T15:19:45Z" level=error msg="Unable to obtain ACME certificate for domains \"*.mydomain.de,mydomain.de\" : unable to generate a certificate for the domains [*.mydomain.de mydomain.de]: acme: Error -> One or more domains had a problem:\n[mydomain.de] error presenting token: netcup: error decoding response of DNS-API, json: cannot unmarshal string into Go struct field ResponseMsg.responsedata of type netcup.ResponseData\n"
```

**Comment by ldez** [comment]
The problem is fixed in Lego https://github.com/xenolf/lego/pull/685

**Comment by ldez** [comment]
And already merged into Traefik #4094, it will be a part of 1.7.4

**Comment by dduportal** [comment]
We are closing this issue since the fix is merged.
Feel free to re-open it if the upcoming 1.7.4 release is not working as expected.
